### PR TITLE
Update github action for onnxruntime sync to open autoPR

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Add better handling of sync action and set this to weekly cron while also opening up PR.

Adds PR labels and reviews for sync action

### CODE REVIEW

File: .github/workflows/sync-onnxrt-main.yaml [hunk 1]
```diff
@@ -1,46 +1,61 @@
...
+  createPullRequest:
+    name: Update and create pull request
+    runs-on: self-hosted
     steps:
-      - uses: actions/checkout@v3
-        with:
-          ref: develop
-          path: ../
+      - name: cleanup-old-runs
+        run: rm -rf onnxruntime
```
Comment (causten) [inline .github/workflows/sync-onnxrt-main.yaml:12]: I think there is a better way without having to rm and clone.  

git ls-remote https://github.com/microsoft/onnxruntime.git HEAD | awk '{ print $1}'

That will print just the sha value so you can direct that to where you need it

File: .github/workflows/sync-onnxrt-main.yaml [hunk 2]
```diff
@@ -1,46 +1,61 @@
...
 
-  update_file:
-    needs: [sync get_date]
-    steps:
-      - run: git clone https://github.com/microsoft/onnxruntime.git && cd onnxruntime && git rev-parse HEAD >> ../test/onnx/.onnxrt-commit
+      - name: get-onnxrt-main
+        run: git clone https://github.com/microsoft/onnxruntime.git
+        
+      - name: extract_sha1 
+        run: cd onnxruntime && echo onnxsha="$(git rev-parse HEAD)" >> $GITHUB_ENV
```
Comment (causten) [inline .github/workflows/sync-onnxrt-main.yaml:21]: git ls-remote https://github.com/microsoft/onnxruntime.git HEAD | awk '{ print $1}' >> $GITHUB_ENV

File: .github/workflows/sync-onnxrt-main.yaml [hunk 3]
```diff
@@ -1,46 +1,61 @@
...
-  PR:
-    needs: Add_commit
-    steps:
-      - name: GitHub Action for creating Pull Requests
-        uses: devops-infra/action-pull-request@v0.5.3
+      - name: echo_sha1
+        run: echo ${{ env.onnxsha }}
+
+      - name: cleanup-run
+        run: rm -rf onnxruntime
```
Comment (causten) [inline .github/workflows/sync-onnxrt-main.yaml:27]: not needed

**Review by causten — CHANGES_REQUESTED** [review]

File: .github/workflows/sync-onnxrt-main.yaml (hunk 3 above)
Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml]: Done

File: .github/workflows/sync-onnxrt-main.yaml (hunk 2 above)
Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml]: This is a great suggestion. I didn't realize we could do that.

File: .github/workflows/sync-onnxrt-main.yaml (hunk 1 above)
Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml]: Perfect. Done

**Review by causten — APPROVED** [review]

File: .github/workflows/sync-onnxrt-main.yaml [hunk 4]
```diff
@@ -1,46 +1,52 @@
...
+
+            [1]: https://github.com/peter-evans/create-pull-request
+          labels: |
+            onnxruntime
+            dependancies
+            automated 
+          assignees: TedThemistokleous
+          reviewers: TedThemistokleous causten
+          draft: false
+          base: master
```
Comment (umangyadav) [inline .github/workflows/sync-onnxrt-main.yaml:52]: Is `base` base for the branch off which this  new one would be created ? Do you want to make it `develop` or keep it `master` ?

File: .github/workflows/sync-onnxrt-main.yaml [hunk 5]
```diff
@@ -1,46 +1,52 @@
...
-          default_author: github_actions
-          push: true
-  PR:
-    needs: Add_commit
-    steps:
-      - name: GitHub Action for creating Pull Requests
-        uses: devops-infra/action-pull-request@v0.5.3
+      - uses: actions/checkout@v3
+        with: 
+          ref: master
```
Comment (umangyadav) [inline .github/workflows/sync-onnxrt-main.yaml:22]: Is this `master` branch name for onnx-rt ?

Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml:22]: No its for migrahx

File: .github/workflows/sync-onnxrt-main.yaml (hunk 4 above)
Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml:52]: Ill move this to develop

File: .github/workflows/sync-onnxrt-main.yaml [hunk 6]
```diff
@@ -1,46 +1,52 @@
...
+          branch: onnxruntime-sync-${{ env.todays_date }}
+          delete-branch: true
+          title: 'Onnxruntime Weekly Sync ${{ env.todays_date }}'
+          body: |
+            Update CI point for Onnxruntime builds
+            - Updated with changes from ${{ env.todays_date }}
+            - Auto-generated by [create-pull-request][1]
+            - Update .onnxrt-commit to ${{ env.onnxsha }}
+
+            [1]: https://github.com/peter-evans/create-pull-request
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply .github/workflows/sync-onnxrt-main.yaml:44]: That's whats used in the comment. 

image

**Review by umangyadav — APPROVED** [review]

**Pull request merged**